lib/features/email/presentation/email_controller.dart

```diff
@@ -137,21 +142,54 @@ class EmailController extends BaseController with AppLoaderMixin {
   @override
   void onClose() {
     _downloadProgressStateController.close();
+    pageController?.dispose();
     _clearWorker();
     super.onClose();
   }
 
+  void _setCurrentPositionEmailInListEmail() {
+    currentIndexPageView = mailboxDashBoardController.emailsInCurrentMailbox.indexOf(mailboxDashBoardController.selectedEmail.value);
+    if(pageController != null && pageController!.hasClients && pageController!.page?.toInt() != currentIndexPageView) {
+      pageController!.jumpToPage(currentIndexPageView);
+    } else {
+      pageController = PageController(initialPage: currentIndexPageView);
+    }
+    _checkEnableNavigatorPageView();
+  }
+
+  void onPageChanged(int index) {
+    mailboxDashBoardController.selectedEmail.value = mailboxDashBoardController.emailsInCurrentMailbox[index];
+  }
+
+  void _checkEnableNavigatorPageView() {
+    canGetNewerEmail.value = currentIndexPageView > 0;
+    canGetOlderEmail.value = mailboxDashBoardController.emailsInCurrentMailbox.length > 1 && currentIndexPageView < mailboxDashBoardController.emailsInCurrentMailbox.length - 1;
+  }
+
+  void getNewerEmail() {
+    currentIndexPageView = currentIndexPageView - 1;
+    pageController?.jumpToPage(currentIndexPageView);
+  }
+
+  void getOlderEmail() {
+    currentIndexPageView = currentIndexPageView + 1;
+    pageController?.jumpToPage(currentIndexPageView);
+  }
+
   void _initWorker() {
-    emailWorker = ever( mailboxDashBoardController.selectedEmail, (presentationEmail) {
+    emailWorker = ever(mailboxDashBoardController.selectedEmail, (presentationEmail) {
       if (presentationEmail is PresentationEmail) {
-        if (_currentEmailId != presentationEmail.id) {
+        _setCurrentPositionEmailInListEmail();
+      if (_currentEmailId != presentationEmail.id) {
           _currentEmailId = presentationEmail.id;
           _resetToOriginalValue();
           _getEmailContentAction(presentationEmail.id);
           if (!presentationEmail.hasRead) {
             markAsEmailRead(presentationEmail, ReadActions.markAsRead);
           }
-          _getAllIdentities();
+          if(_identitySelected == null) {
```

Comment:
why need check it

Reply:
when changed pageview of email content, email selected will be changed. So we don't need call `_getAllIdentities ` when `_identitySelected` != null